Xamarin.Forms.Platform.Android/FastRenderers/AccessibilityProvider.cs

```diff
@@ -137,19 +136,15 @@ bool SetHint(string hint = GetFromElement)
 				return true;
 			}
 
-			if (_defaultHint == null)
-			{
-				_defaultHint = textView.Hint;
-			}
+			string accessibilityName = (string)Element.GetValue(Accessibility.NameProperty);
+			string accessibilityHint = (string)Element.GetValue(Accessibility.HintProperty);
 
-			string value = hint;
-			if (value == GetFromElement)
+			if (string.IsNullOrWhiteSpace(accessibilityName) && string.IsNullOrWhiteSpace(accessibilityHint))
 			{
-				value = string.Join(". ", (string)Element.GetValue(Accessibility.NameProperty),
-					(string)Element.GetValue(Accessibility.HintProperty));
+				return true;
 			}
 
-			textView.Hint = !string.IsNullOrWhiteSpace(value) ? value : _defaultHint;
+			textView.Hint = string.Join(". ", accessibilityName, accessibilityHint);
```

Comment:
I dont know i i did somethig wrong with commits, but that is the old commit you are looking at. I dont think I can be any clearer what I mean with this PR in the description, i even pasted images.

This is a newer commit where i changed the code ( its included in this PR)
https://github.com/xamarin/Xamarin.Forms/pull/904/commits/09b4ce162e8d18937198a5e1ec204e58be998e68